airflow/serialization/serde.py

```diff
@@ -176,11 +189,11 @@ def deserialize(o: T | None, full=True, type_hint: Any = None) -> object:
     if isinstance(o, _primitives):
         return o
 
-    if isinstance(o, _iterables):
+    if isinstance(o, _collections):
         return [deserialize(d) for d in o]
 
     if not isinstance(o, dict):
-        raise TypeError()
+        raise TypeError("not deserializable")
```

Comment:
It is probably better to explain what type we are getting if we are improving the error message,.